src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -977,9 +1004,11 @@ const updateTransactions = (
         tx: processedTx
       })
 
-      // Add task to grab transactionSpecific payload
-      taskCache.updateTransactionSpecificCache[txId] = {
-        processing: false
+      if (needsTxSpecific) {
```

Comment:
Should we inline the condition here rather than thread through a parameter?

Reply:
Nah, we'll keep it wired through. There's an ongoing repayment of tech-debt to make these arg-bag parameters saner.